module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -282,15 +289,31 @@ public function preRequest(\Laminas\Http\Headers $headers, $params)
      */
     protected function renewTenantToken()
     {
+        // If not using legacy authentication, see if the token has expired before trying to renew it
+        if (! $this->useLegacyAuthentication() && ! $this->checkTenantTokenExpired()) {
+            $currentTime = gmdate('D, d-M-Y H:i:s T', strtotime('now'));
+            $this->debug(
+                'No need to renew token; not yet expired. ' . $currentTime . ' < ' . $this->tokenExpiration .
+                'Username: ' . $this->config['API']['username'] . ' Token: ' . substr($this->token, 0, 30) . '...'
+            );
+            return;
+        }
+        $startTime = microtime(true);
         $this->token = null;
         $response = $this->performOkapiUsernamePasswordAuthentication(
             $this->config['API']['username'],
             $this->getSecretFromConfig($this->config['API'], 'password')
         );
         $this->token = $this->extractTokenFromResponse($response);
+        $this->tokenExpiration =
+            $this->useLegacyAuthentication() ?
+            null : $this->extractTokenExpirationFromResponse($response);
```

Comment:
Just a minor stylistic suggestion (for consistency with more existing code):

```suggestion
        $this->tokenExpiration = $this->useLegacyAuthentication()
            ? null : $this->extractTokenExpirationFromResponse($response);
```

Reply:
Thanks -- I was looking for an example when writing it to find the preferred style and wasn't finding one.